controllers/agents/app/servicebinding_controller.go

```diff
@@ -642,11 +642,12 @@ func (r *ServiceBindingReconciler) unbindApplications(ctx context.Context,
 
 func verifyApplicationSatisfiesServiceBindingSpec(obj *unstructured.Unstructured, sb v1alpha1.ServiceBinding) bool {
 	switch {
-	case sb.Spec.Application.Name == obj.GetName():
+	case sb.Spec.Application.Selector.ByName == obj.GetName():
 		return true
-	case sb.Spec.Application.Selector != nil:
+	case sb.Spec.Application.Selector.ByLabels != nil:
+		// TODO: handle sb.Spec.Application.Selector.ByLabels.MatchExpressions
 		for label, value := range obj.GetLabels() {
-			val, ok := sb.Spec.Application.Selector.MatchLabels[label]
+			val, ok := sb.Spec.Application.Selector.ByLabels.MatchLabels[label]
 			return ok && value == val
 		}
```

Comment:
On inspection, this for loop seems very fishy to me.  We begin iterating, but always short circuit and return on the first iteration?  What's going on here?

Reply:
updated